harness/determined/pytorch/deepspeed/_deepspeed_trial.py

```diff
@@ -0,0 +1,1054 @@
+import logging
+import pathlib
+import pickle
+import random
+import time
+from abc import abstractmethod
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Type, Union, cast
+
+import deepspeed
+import numpy as np
+import torch
+
+import determined as det
+from determined import layers, pytorch, util, workload
+from determined.common import check, storage
```

Comment:
please, don't even import the `check` module in new code.  For validating user inputs, use if statements.  For validating internal assumptions or mypy-related cases, use asserts.